### PR TITLE
http: Add a method for returning a 404 with data

### PR DESCRIPTION
**Pull request by briankassouf**

### CODE REVIEW

File: logical/response.go [hunk 1]
```diff
@@ -130,3 +132,23 @@ func ListResponseWithInfo(keys []string, keyInfo map[string]interface{}) *Respon
 
 	return resp
 }
+
+// Respond404WithData takes a response and converts it to a raw response with a
+// 404 Status Code.
+func Respond404WithData(resp *Response, reqID string) (*Response, error) {
```
Comment (vishalnayak) [inline logical/response.go]: Do you see value in making this `RespondWithStatusCode` and parameterizing the 404?

File: http/logical.go [hunk 2]
```diff
@@ -183,7 +183,7 @@ func respondLogical(w http.ResponseWriter, r *http.Request, req *logical.Request
 // respondRaw is used when the response is using HTTPContentType and HTTPRawBody
 // to change the default response handling. This is only used for specific things like
 // returning the CRL information on the PKI backends.
-func respondRaw(w http.ResponseWriter, r *http.Request, resp *logical.Response) {
+func respondRaw(w http.ResponseWriter, r *http.Request, req *logical.Request, resp *logical.Response) {
```
Comment (calvn) [inline http/logical.go]: Where is `req` being used?

File: http/logical.go [hunk 3]
```diff
@@ -258,6 +258,7 @@ func respondRaw(w http.ResponseWriter, r *http.Request, resp *logical.Response)
 			}
 		case []byte:
 			body = bodyRaw.([]byte)
+
```
Comment (calvn) [inline http/logical.go]: Extra newline here

File: logical/response.go [hunk 4]
```diff
@@ -130,3 +131,23 @@ func ListResponseWithInfo(keys []string, keyInfo map[string]interface{}) *Respon
 
 	return resp
 }
+
+// RespondWithStatusCode takes a response and converts it to a raw response with
+// the provided Status Code.
+func RespondWithStatusCode(resp *Response, reqID string, code int) (*Response, error) {
```
Comment (jefferai) [inline logical/response.go]: Rather than pass in the req ID, I suggest just passing in the request...same number of parameters, but a little more obvious.

Comment (briankassouf) [reply logical/response.go]: Done!

**Pull request merged**